modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,422 @@
+      next unless r.class == Net::DNS::RR::SOA
+      records << r.mname
+      report_host(host: r.mname, info: 'SOA')
+      vprint_good("#{domain}: SOA: #{r.mname}")
     end
+    return if records.none?
+    save_loot('ENUM_SOA', 'text/plain', domain, "#{records.join(',')}", domain)
+    records
+  end
+
+  def get_txt(domain)
+    resp = dns_query(domain, 'TXT')
+    return if resp.blank? || resp.answer.blank?
 
-    if(datastore['ENUM_TLD'])
-      tldexpnd(datastore['DOMAIN'],datastore['NS'])
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::TXT
+      records << r.txt
+      vprint_good("#{domain}: TXT: #{r.txt}")
     end
+    return if records.none?
+    save_loot('ENUM_TXT', 'text/plain', domain, "#{records.join(',')}", domain)
+    records
+  end
 
-    if(datastore['ENUM_BRT'])
-      if wldcrd and datastore['STOP_WLDCRD']
-        print_error("Wildcard record found!")
-      else
-        dnsbrute(datastore['DOMAIN'],datastore['WORDLIST'],datastore['NS'])
-      end
+  def get_tld(domain)
+    domain_ = domain.split('.')
+    domain_.pop
+    domain_ = domain_.join('.')
+
+    tlds = [
+      'com', 'org', 'net', 'edu', 'mil', 'gov', 'uk', 'af', 'al', 'dz',
+      'as', 'ad', 'ao', 'ai', 'aq', 'ag', 'ar', 'am', 'aw', 'ac', 'au',
+      'at', 'az', 'bs', 'bh', 'bd', 'bb', 'by', 'be', 'bz', 'bj', 'bm',
+      'bt', 'bo', 'ba', 'bw', 'bv', 'br', 'io', 'bn', 'bg', 'bf', 'bi',
+      'kh', 'cm', 'ca', 'cv', 'ky', 'cf', 'td', 'cl', 'cn', 'cx', 'cc',
+      'co', 'km', 'cd', 'cg', 'ck', 'cr', 'ci', 'hr', 'cu', 'cy', 'cz',
+      'dk', 'dj', 'dm', 'do', 'tp', 'ec', 'eg', 'sv', 'gq', 'er', 'ee',
+      'et', 'fk', 'fo', 'fj', 'fi', 'fr', 'gf', 'pf', 'tf', 'ga', 'gm',
+      'ge', 'de', 'gh', 'gi', 'gr', 'gl', 'gd', 'gp', 'gu', 'gt', 'gg',
+      'gn', 'gw', 'gy', 'ht', 'hm', 'va', 'hn', 'hk', 'hu', 'is', 'in',
+      'id', 'ir', 'iq', 'ie', 'im', 'il', 'it', 'jm', 'jp', 'je', 'jo',
+      'kz', 'ke', 'ki', 'kp', 'kr', 'kw', 'kg', 'la', 'lv', 'lb', 'ls',
+      'lr', 'ly', 'li', 'lt', 'lu', 'mo', 'mk', 'mg', 'mw', 'my', 'mv',
+      'ml', 'mt', 'mh', 'mq', 'mr', 'mu', 'yt', 'mx', 'fm', 'md', 'mc',
+      'mn', 'ms', 'ma', 'mz', 'mm', 'na', 'nr', 'np', 'nl', 'an', 'nc',
+      'nz', 'ni', 'ne', 'ng', 'nu', 'nf', 'mp', 'no', 'om', 'pk', 'pw',
+      'pa', 'pg', 'py', 'pe', 'ph', 'pn', 'pl', 'pt', 'pr', 'qa', 're',
+      'ro', 'ru', 'rw', 'kn', 'lc', 'vc', 'ws', 'sm', 'st', 'sa', 'sn',
+      'sc', 'sl', 'sg', 'sk', 'si', 'sb', 'so', 'za', 'gz', 'es', 'lk',
+      'sh', 'pm', 'sd', 'sr', 'sj', 'sz', 'se', 'ch', 'sy', 'tw', 'tj',
+      'tz', 'th', 'tg', 'tk', 'to', 'tt', 'tn', 'tr', 'tm', 'tc', 'tv',
+      'ug', 'ua', 'ae', 'gb', 'us', 'um', 'uy', 'uz', 'vu', 've', 'vn',
+      'vg', 'vi', 'wf', 'eh', 'ye', 'yu', 'za', 'zr', 'zm', 'zw', 'int',
+      'gs', 'info', 'biz', 'su', 'name', 'coop', 'aero']
+
+    records = []
+    tlds.each do |tld|
+      tldr = get_a("#{domain_}.#{tld}")
```

Comment:
This results in an exception for me when trying the `.co` TLD for my target domain because for whatever reason the CNAME  is 66 characters which is greater than the 63 allowed by the RFC:

```
msf auxiliary(enum_dns) > run

[-] Auxiliary failed: ArgumentError Label data cannot exceed 63 chars
[-] Call stack:
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/names/names.rb:50:in `pack_name'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/rr/cname.rb:31:in `build_pack'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/rr.rb:335:in `new_from_binary'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/rr.rb:176:in `parse_packet'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/rr.rb:327:in `new_from_binary'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/rr.rb:176:in `parse_packet'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/packet.rb:522:in `block in new_from_data'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/packet.rb:521:in `times'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/packet.rb:521:in `new_from_data'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/packet.rb:153:in `parse'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/resolver.rb:986:in `send'
[-]   /home/jhart/rapid7/metasploit-framework/lib/net/dns/resolver.rb:894:in `query'
[-]   /home/jhart/rapid7/metasploit-framework/modules/auxiliary/gather/enum_dns.rb:98:in `dns_query'
[-]   /home/jhart/rapid7/metasploit-framework/modules/auxiliary/gather/enum_dns.rb:208:in `get_a'
[-]   /home/jhart/rapid7/metasploit-framework/modules/auxiliary/gather/enum_dns.rb:334:in `block in get_tld'
[-]   /home/jhart/rapid7/metasploit-framework/modules/auxiliary/gather/enum_dns.rb:333:in `each'
[-]   /home/jhart/rapid7/metasploit-framework/modules/auxiliary/gather/enum_dns.rb:333:in `get_tld'
[-]   /home/jhart/rapid7/metasploit-framework/modules/auxiliary/gather/enum_dns.rb:73:in `run'
[*] Auxiliary module execution completed
```

This means that the SRV enumeration won't run because it happens after the TLD checking.  You should probably add better error handling around any of the calls that go out to `Net::DNS`



```